### DISCUSSION TITLE
Posts do not show on mastodon

### DISCUSSION THREAD

**Discussion by PullingBerlin** [Automattic/wordpress-activitypub#545] [General]
Hi, I might have a similar problem as described [here](https://github.com/Automattic/wordpress-activitypub/discussions/537). I’m using WordPress with Strato and activated the ActivityPub plugin. I am able to follow my wp profile on mastodon (I find it when searching for it). However, I do not see any posts. I see the bio description from wp and it also shows the number of already published wp posts on mastodon (this is the number of posts the profile had when I started following it on mastodon, it doesn't change anymore, even when I unfollow and follow again).

my wp page: http://electricmonk.de/
Profile URL: http://electricmonk.de/@allofit
Mastodon: @Benni@det.social

Any ideas anyone? I would really appreciate your help!

**Comment by pfefferle** [comment]
Hey @PullingBerlin can you please check the WP_Cron queue? Is this processes correctly?

**Reply by pfefferle** [reply]
It seems to work fine if I copy and paste the URL of a Post into the search field, so I assume that something with the publishing/queuing process is broken.

<img width="592" alt="Screenshot 2023-11-07 at 10 09 01" src="https://github.com/Automattic/wordpress-activitypub/assets/2373/39896f22-f1c4-49e5-95c1-f0fe68cc52c4">

**Reply by PullingBerlin** [reply]
Hi @pfefferle, Cron Jobs seem to work:

![Screenshot from 2023-11-11 07-24-26](https://github.com/Automattic/wordpress-activitypub/assets/74929809/b61f0ec1-c78e-4d1c-b67f-0132434f3544)

**Reply by PullingBerlin** [reply]
Strange, that when pasting the URL the post is shown (and remains).

**Comment by lordmatt** [comment]
I had an issue similar to this. It turned out my host was using an aggressive anti-DDoS system that saw ActivityPub requests as an attack. I spoke to the tech team and we found a workaround which was just to dial back the protection on the ActivityPub domain. I was going spare trying to figure out what the problem was before then.

**Reply by PullingBerlin** [reply]
Thanks @lordmatt, maybe it's something like this. I contacted Strato and gave them this hint. I'm not very optimistic that they will do anything. I'm using a WordPress package and do not have admin access to the server itself...

**Reply by PullingBerlin** [reply]
So of course, Strato answered that external plugins are not their responsibility and I should ask the developer of the plugin...so apparently no way to get the plugin to work in some hosted/packaged WP-scenarios.

**Comment by trozman** [comment]
Hi @pfefferle, great plugin!
Nevertheless, I have similar issues (not seeing updates in Mastodon):
- installed a fresh WP, subdomain etc (no caching enabled), here: [](https://blog.rozman.info)
- installed activitypub plugin (and other recommended plugins by activitypub, e.g. friends) 
- I can follow my WP user from Mastodon or Calckey (user: https://blog.rozman.info/author/tomi/ or @tomi@blog.rozman.info)
- I don't see updates (new posts) in Mastodon under WP user after I publish posts
- I see new posts in Mastodon ONLY if I manually search for a post from Mastodon using URL

What I did until now:
- checked mod_evasive module in Apache and whitelisted my reverse proxy (which triggered false DoS attack)
- checked wp_cron queue, result: I can see activitypub cron jobs, which clear without error messages
- enabled WP DEBUG and checked the debug.log file, which show requests to various fediverse servers (followers of my blog). I found some errors in it:

```
[04-Dec-2023 09:55:02 UTC] **Request to: https://mastodon.social/inbox with response: WP_Error Object**
(
    [errors] => Array
        (
            [401] => Array
                (
                    [0] => Failed HTTP Request
                )

        )
```
...

Any idea? What else should I do to see updates? Thanks!

**Reply by pfefferle** [reply]
401 says that Mastodon does not like the signature. Have you experimented with the signature recently or done something that might have changed the public/private key and or forced a re-creation?

**Reply by trozman** [reply]
Thanks for the feedback! 
Not sure, I think I renewed the Letsencrypt subdomain certificate on my reverse proxy nginx server  (which redirects traffic to WP server) several days ago.

**Reply by pfefferle** [reply]
that should be no problem, I guess... It's more about the keys generated by the plugin to sign the requests to Mastodon et al.

**Reply by trozman** [reply]
I've just found the solution! It was a misconfigured reverse proxy - I forgot to include certificates to nginx configuration file for the new subdomain :)
Nevertheless, thanks for the direction - without your tip it would take me weeks to find it. 
Now it works!

**Reply by pfefferle** [reply]
nice! it's usually the stupid questions that put you on the right track :)

**Reply by pfefferle** [reply]
And with "stupid questions" I meant mine, not yours! 😳

**Reply by trozman** [reply]
Unfortunately, as it seems the luck didn't last long. Sometimes it federates, mostly not (...Failed HTTP Requests [401] in debug.log). I can't find the pattern.
